src/arc.ts

```diff
@@ -132,34 +161,45 @@ export async function initializeArc(provider?: any): Promise<boolean> {
     const readonly = typeof provider === "string";
 
     // if there is no existing arc, we create a new one
-    if ((window as any).arc) {
-      arc = (window as any).arc;
+    if ((window as any).arcs[network]) {
+      arc = (window as any).arcs[network];
       arc.web3 = new Web3(provider);
-    } else {
+    }
+    else {
       arc = new Arc(arcSettings);
     }
 
     let contractInfos;
-    if (USE_CONTRACTINFOS_CACHE) {
-      contractInfos = require(`data/contractInfos-${targetedNetwork()}.json`);
-      arc.setContractInfos(contractInfos);
-    } else {
-      try {
-        contractInfos = await arc.fetchContractInfos();
-      } catch (err) {
-        // eslint-disable-next-line no-console
-        console.error(`Error fetching contractinfos: ${err.message}`);
+
+    try {
+      contractInfos = await arc.fetchContractInfos();
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(`Error fetching contractinfos: ${err.message}`);
```

Comment:
We should rather do something here. Let's discuss what should happen here

Reply:
What can we do if not throwing an error?